src/meta/src/barrier/checkpoint/creating_job/mod.rs

```diff
@@ -158,10 +159,170 @@ impl CreatingStreamingJobControl {
             },
             upstream_lag: GLOBAL_META_METRICS
                 .snapshot_backfill_lag
-                .with_guarded_label_values(&[&table_id_str]),
+                .with_guarded_label_values(&[&format!("{}", job_id)]),
+        })
+    }
+
+    fn resolve_upstream_log_epochs(
+        snapshot_backfill_upstream_tables: &HashSet<TableId>,
+        upstream_table_log_epochs: &HashMap<TableId, Vec<Vec<u64>>>,
```

Comment:
Feel like you mentioned this before offline. But I can't quite recall the meaning of `Vec<Vec<u64>>`.

Can you separate it into a separate struct or type alias with some docstrings? Didn't get the purpose of nesting it.